roles/openshift_grafana/tasks/main.yaml

```diff
@@ -0,0 +1,72 @@
+---
+- name: Create grafana namespace
+  command: oc create namespace grafana
+
+- name: Switch to grafana project
+  command: oc project grafana
+
+- name: Templatize grafana yaml
+  template: src=grafana-ocp.yaml dest=/tmp/grafana-ocp.yaml
+
+- name: load the grafana taml
+  command: oc new-app -f /tmp/grafana-ocp.yaml
+
+- name: Wait to grafana be running
+  command: oc get pod |grep grafana |awk '{print $3}'
+  register: grafana_pod
+  failed_when: grafana_pod.stdout = 'ImgError'
+  retries: 12
+  delay: 5
+  until: grafana_pod.stdout == 'Running'
+
+- name: Get oc token
+  command: oc sa get-token prometheus -n kube-system
+  register: token
+
+- name: Get oc route
+  command: oc get route |grep grafana |awk '{print $2}'
+  register: route
+
+- name: Get prometheus server
+  command: oc get route --all-namespaces |grep prometheus |awk '{print $3}'
+  register: prometheus
+
+- name: Register token
+  var:
+    ds_json: "{\"name\":\""{{ gf_ds_name }}"\",\"type\":\"prometheus\",\"typeLogoUrl\":\"\",\"access\":\"proxy\",\"url\":\"https://"{{ prometheus }}"\",\"basicAuth\":false,\"withCredentials\":false,\"jsonData\":{\"tlsSkipVerify\":true,\"token\":\"{{ token }}\"}}"
```

Comment:
You probably want to save this in `defaults/main.yml`. You can use the `to/from_json` type filters here to make it a datastructure or serialize it.